cpp/CMakeLists.txt

```diff
@@ -341,6 +347,32 @@ add_library(
   src/distance/detail/pairwise_matrix/dispatch_russel_rao_float_float_float_int.cu
   src/distance/detail/pairwise_matrix/dispatch_russel_rao_half_float_float_int.cu
   src/distance/detail/pairwise_matrix/dispatch_russel_rao_double_double_double_int.cu
+  src/distance/detail/pairwise_matrix/dispatch_canberra_double_double_double_int64_t.cu
+  src/distance/detail/pairwise_matrix/dispatch_canberra_float_float_float_int64_t.cu
+  src/distance/detail/pairwise_matrix/dispatch_correlation_double_double_double_int64_t.cu
```

Comment:
Question- do we need all of these to support int64_t or just the euclidean? Kmeans is inherently coupled to euclidean (and cosine, though that's not an option yet in our kmeans).

Reply:
pushed out a change to limit to just l2expanded distance w/ int64_t indices here https://github.com/rapidsai/cuvs/pull/348/commits/63c11ce771f2b6b319d846e1e3c154b7baa6cdf8 - lmk what you think